library/src/meta/reference.rs

```diff
@@ -160,6 +161,13 @@ impl Show for RefElem {
             }
 
             let elem = elem.at(span)?;
+
+            if elem.func() == FootnoteElem::func() {
+                let mut elem = FootnoteElem::new_reference(target);
+                elem.synthesize(vt, styles)?;
```

Comment:
is this synthesis necessary? probably, since you wouldn't have added it otherwise. but I would have thought that it is taken care of automatically.

Reply:
I'm also wondering how many compilations footnote references incur. Citations are carefully implemented such that only two compilations are required in the common case. It _should_ be the same here if I'm not mistaken, but it would be good to check that.

Reply:
> is this synthesis necessary?

After looking at it for a bit, I don't think it is necessary (and all tests pass without it). To be honest, I think I called `synthetize` here by analogy to how it is called in `RefElem::to_citation`. I am still new to the codebase, so I'm sorry for writting stupid things sometimes.

> I'm also wondering how many compilations footnote references incur.

If you are talking about the number of iterations of the loop in `model::typeset`, regular footnotes take two compilations both before and after my change, but references to footnotes require a third cycle.

Reply:
That's not stupid at all! It's very intricate and I'm actually myself not entirely sure why to_citation calls it. The whole synthesis thing is a bit weird.

I wonder whether references to footnotes could work without a third cycle. In theory, all necessary information should be available after the first compilation. But it's not that important, we can look into that later.

Reply:
I also think two cycles is a reachable minimum, at least if you don't have any forward reference to a footnote. But I don't think I would be able to find what is causing a third cycle in the current implementation, and how to fix it.

I noticed that replacing
```rust
return Ok(FootnoteElem::with_label(target).pack().spanned(span));
```
with
```rust
return Ok(FootnoteElem::with_label(target).show(vt, styles)?.spanned(span));
```
only uses two cycles, but then of course reference footnotes are not styled properly.